feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/detail/BalanceDetailViewModel.kt

```diff
@@ -126,6 +136,30 @@ class BalanceDetailViewModel(
         .inBackground()
         .share()
 
+    val priceChartTitle = assetFlow.map {
+        val tokenName = it.token.configuration.symbol.value
+        resourceManager.getString(R.string.price_chart_title, tokenName)
+    }
+
+    val priceChartFormatters: Flow<PriceChartTextInjectors> = flowOf {
+        val currency = currencyInteractor.getSelectedCurrency()
+
+        PriceChartTextInjectors(
+            RealPricePriceTextInjector(currency),
+            RealPriceChangeTextInjector(resourceManager, currency),
+            RealDateChartTextInjector(resourceManager)
+        )
+    }
+
+    private val priceCharts: Flow<List<AssetPriceChart>?> = assetFlow.flatMapLatest {
+        val priceId = it.token.configuration.priceId ?: return@flatMapLatest flowOf { null }
+        chartsInteractor.getCharts(priceId)
+    }
+
+    val priceChartModels = priceCharts.map { charts ->
+        charts?.map { mapChartsToUi(it) }
+    }
```

Comment:
Please do not forget to always do `shareInBackground` on flows that do non-trivial computation